aten/src/ATen/core/dispatch/Dispatcher.cpp

```diff
@@ -112,14 +81,34 @@ RegistrationHandleRAII Dispatcher::registerDef(FunctionSchema schema) {
 
   OperatorName op_name = schema.operator_name();
 
-  auto op = findOrRegisterSchema_(std::move(schema));
+  const auto found = findSchema(op_name);
+  OperatorHandle op = [&] {
+    if (found == c10::nullopt) {
+      // Initial def
+      operators_.emplace_back(std::move(schema));
+      OperatorHandle op(--operators_.end());
+      operatorLookupTable_.write([&] (ska::flat_hash_map<OperatorName, OperatorHandle>& operatorLookupTable) {
+        operatorLookupTable.emplace(op_name, op);
+      });
+      // Call the hook after the operator is setup!
+      listeners_->callOnOperatorRegistered(op);
+      return op;
+    } else {
+      // Re-registration for BC reasons, test for consistency
```

Comment:
This is not only for bc reasons, right? It also happens when there was already an impl() but no def() yet and now a def() is coming in. Right? I think the TORCH_CHECK below is the only line that is there for bc reasons only.

Reply:
yes. I've fixed this already in a later patch